File: DSA/sorting/selection_sort.cpp
#include <iostream>

using namespace std;

void selection_sort(int array[], int len){
    for(int i=0;i<len;i++){
        int min = i;
        for(int j=i;j<len;j++){
            if (array[min]>array[j]){
                min = j;
            }
        }

        int temp = array[min];
        array[min] = array[i];
        array[i] = temp;
    }
}


void printarray(const int array[], int len){
    for ( int i=0; i<len;i++){
        cout << array[i] << " ";
    }
    cout << endl;
}

int main(){
    int len = 11;
    int array [len] = {10,9,8,7,6,5,4,3,2,1,0};

    selection_sort(array, len);

    printarray(array, len);

    return 0;
}

File: DSA/sorting/selection_sort.py
def selection_sort(array):

    for i in range(len(array)):
        min = i
        for j in range(i, len(array)):
            if array[j] < array[min]:
                min = j

        array[i], array[min] = array[min], array[i]

    return array
            

if __name__ == '__main__':
    # Test
    test_list = [50, 46, 29, 103, 50302, 1]
    print(selection_sort(test_list))